codec-http/src/main/java/io/netty/handler/codec/http/HttpServerUpgradeHandler.java

```diff
@@ -313,32 +313,32 @@ private boolean upgrade(final ChannelHandlerContext ctx, final FullHttpRequest r
 
         // Create the user event to be fired once the upgrade completes.
         final UpgradeEvent event = new UpgradeEvent(upgradeProtocol, request);
+        final ChannelFuture writeComplete = ctx.writeAndFlush(upgradeResponse);
```

Comment:
write and flush *shouldn't* throw ... but consider including it in the `try` block below just to have less assumptions about releasing the `event` relative to behavior of this method.